tests/lib/external/snapd-testing-tools/remote/remote.wait-for

```diff
@@ -110,6 +112,24 @@ wait_for_reboot() {
         return
     fi
 
+    if [ -n "$expect_passphrase" ]; then
+        if [ ! -f "$log_file" ]; then
+            echo "remote.wait-for --expect-passphrase without passing an existing --log-file"
+            return 1
+        fi
+        while [ "$attempts" -ge 0 ]; do
+            echo -n '.'
+            attempts=$(( attempts - 1 ))
+            # Wait for passphrase prompt from serial log file.
+            if MATCH "Please enter the passphrase" < ${log_file}; then
+                # Enter passphrase to continue boot.
+                echo "$expect_passphrase" | netcat -N localhost 7777
```

Comment:
If we use bash, maybe we can use `/dev/tcp/localhost/7777` instead of relying on an external tool that has lots of forks and different command lines.

Or even better, switch that to a unix socket.

Reply:
I'd check how far back support for tcp sockets in bash goes, so that we're not surprised by some old version

Reply:
I was not able to get it working with tcp sockets, even locally. I couldn't get qemu to understand that connection is closed and interpret a `\n` or `\r` to enter the passphrase.